src/compile/compile.ts

```diff
@@ -88,11 +93,18 @@ function assembleTopLevelModel(model: Model, topLevelProperties: TopLevelPropert
     return true;
   });
 
+  if (autosize.type === 'fit') {
```

Comment:
This logic should be consolidate as a part of `normalizeAutoSize` and `normalizeAutoSize` should be called only once in `compile.ts`.